### DISCUSSION TITLE
Companion DAP package

### DISCUSSION THREAD

**Discussion by cyruseuros** [joaotavora/eglot#718] [Ideas]
To the best of my knowledge, the DAP ofter relies on information from the LSP for the same language, so it would make sense to have an eglot companion package to achieve this interop.

**Comment by cyruseuros** [comment]
Note that I'm raising this issue in light of [this](https://github.com/emacs-lsp/dap-mode/issues/2) response on the `dap-mode` repo, and the fact that the existence of dap-mode seems to be enough of a reason for `realgud` to [not implement this feature](https://github.com/realgud/realgud/issues/231). Nothing against these projects, people are not obliged to volunteer their time, but if Eglot ends up in Emacs core it would be good to have core module implementing the DAP as well.

**Comment by joaotavora** [comment]
Yes, I think this is a nice idea.  

But it should not be a part of Eglot, rather its own package.  

The `jsonrpc.el` package (that is already in Emacs core), is probably quite useful for developing such a package, since as far as I can tell it's a JSONRPC protocol as well.  As probably would be many LSP-agnostic functions and utilities that currently live in `eglot.el`.  

You can make that future `gudap.el` (name idea) package depend on `eglot.el` (as a library) for those utilities.  Or, better yet, or extract them into `jsonrpc.el`, so that `gudap.el` depends only on them.

I would also reuse as much of `gud.el` as possible, and add to it when the needed functionality isn't there yet.

**Reply by joaotavora** [reply]
Also, converted this into a discussion.  Issues in Eglot are for issues created with its issue-creating template.

**Comment by hab25** [comment]
https://github.com/svaante/dape
